### PR TITLE
TST: Rewrite the reproducibility tests

### PR DESCRIPTION
**Pull request by caspervdw**
As raised in #459 and #461 by @charlesreid1 , the current reproducibility tests should be more modular and should be adapted to the changes in the current tracking algorithms. I redid the tests, splitting the functionality in the following:

 - feature finding (grey dilation) without preprocessing
 - feature finding (grey dilation) with preprocessing
 - refinement of given features
 - the `locate` procedure
 - linking with memory
 - linking without memory
 - characterization

I think this should be the last PR before the release of v0.4

### CODE REVIEW

**Comment by caspervdw** [comment]
I (finally) generated and pushed the data for the pre-scipy-1.0 reproducibility tests. As discussed before in #459, the rounding changes in `scipy.ndimage.filters.uniform_filter1d` lead to minor changes in our preprocessed images. These propagate all the way into the outcome of `find`, `refine` and `characterize`. The fact that the test images are very noisy makes them susceptible to these subtle changes in preprocessing. This is a good thing, it allows us to catch these kind of changes. But it requires separate reference data for pre- and post-scipy 1.0 reproducibility tests.

**Comment by caspervdw** [comment]
I additionally tackled a `FutureWarning` that was triggered by `np.issubdtype(arg, np.int)` or `np.issubdtype(arg, np.float)`. Adapted these to abstract types `np.integer` and `np.floating`, respectively.

**Comment by caspervdw** [comment]
@danielballan @tacaswell @nkeim  This PR is ready for merging

File: trackpy/tests/test_reproducibility.py [hunk 1]
```diff
@@ -63,78 +73,122 @@ class TestReproducibility(StrictTestCase):
...
+        try:
+            assert_allclose(actual_coords,
+                            self.expected_characterize[:, :2])
+        except AssertionError:
+            raise AssertionError('The characterize tests failed as the coords'
+                                 ' found by locate were not reproduced.')
+        assert_allclose(actual_char,
+                        self.expected_characterize[:, 2:])
 
 # SCRIPT TO GENERATE THE FEATURES
```
Comment (danielballan) [inline trackpy/tests/test_reproducibility.py]: Can this be put somewhere where it's possible to run it programmatically?

**Comment by caspervdw** [comment]
@danielballan  Done

**Comment by danielballan** [comment]
Thanks, @caspervdw. This is a solid improvement.

**Pull request merged**